src/Eloquent/DocumentModel.php

```diff
@@ -117,7 +117,9 @@ public function fromDateTime($value): UTCDateTime
         if (! $value instanceof DateTimeInterface) {
             $value = parent::asDateTime($value);
         }
-
+        if (isset($this->dateFormat)) {
```

Comment:
This change should be reverted. We don't want to return anything different than a UTCDateTime instance.